utils/tuple_utils.hh

```diff
@@ -0,0 +1,27 @@
+#include <functional>
+#include <tuple>
+
+namespace utils {
+    // Insert a new item of type NewItem into SourceTuple to produce ResultTuple.
+    // ResultTuple mirrors SourceTuple, with the inclusion of an extra element of type NewItem.
+    // SourceTuple and ResultTuple can be arbitrary specialisations of std::tuple or rpc::tuple.
+    template<typename ResultTuple, typename SourceTuple, typename NewItem>
+    requires (std::tuple_size_v<SourceTuple> + 1 == std::tuple_size_v<ResultTuple>)
+    static ResultTuple tuple_insert(SourceTuple&& source_tuple, NewItem&& new_item) {
+        constexpr auto source_size = std::tuple_size_v<SourceTuple>;
+        constexpr auto result_size = std::tuple_size_v<ResultTuple>;
+        constexpr auto item_index = std::invoke([]<std::size_t... Is>(std::index_sequence<Is...>) {
+            auto result = result_size;
+            (void) ((std::is_same_v<std::tuple_element_t<Is, ResultTuple>, NewItem> && (result = Is, true)) || ...);
```

Comment:
This depends on types not repeating, which is fragile for a general utility, but ok.